src/infer_request.cc

```diff
@@ -133,6 +133,38 @@ InferRequest::ShmHandle()
   return shm_handle_;
 }
 
+void
+InferRequest::SetPrevPromise(
```

Comment:
Why do we need promises for the inference request?

Reply:
We would need to store the previous promise so that we can call `set_value` later with the current response.

Reply:
This should not be a problem anymore by using `infer_payload` object.